File: app/src/main/java/com/juniperphoton/myerlistandroid/util/TypefaceUtil.java
package com.juniperphoton.myerlistandroid.util;


import android.content.Context;
import android.graphics.Typeface;
import android.widget.TextView;

import java.util.HashMap;

public class TypefaceUtil {
    private static HashMap<String, Typeface> typefaceHashMap;

    public static void setTypeFace(TextView textView, String path, Context context) {
        if (typefaceHashMap == null) {
            typefaceHashMap = new HashMap<>();
        }
        Typeface typeface = typefaceHashMap.get(path);
        if (typeface == null) {
            typeface = Typeface.createFromAsset(context.getAssets(), path);
            typefaceHashMap.put(path, typeface);
        }
        textView.setTypeface(typeface);
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/util/TypefaceUtil.kt
package com.juniperphoton.myerlist.util

import android.content.Context
import android.graphics.Typeface
import android.widget.TextView

import java.util.HashMap

object TypefaceUtil {
    private var typefaceHashMap: HashMap<String, Typeface>? = null

    fun setTypeFace(textView: TextView, path: String, context: Context) {
        if (typefaceHashMap == null) {
            typefaceHashMap = HashMap<String, Typeface>()
        }
        var typeface: Typeface? = typefaceHashMap!![path]
        if (typeface == null) {
            typeface = Typeface.createFromAsset(context.assets, path)
            typefaceHashMap!!.put(path, typeface)
        }
        textView.typeface = typeface
    }
}